mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -171,25 +171,38 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
   @Since("2.3.0")
   def setOutputCols(value: Array[String]): this.type = set(outputCols, value)
 
-  private[feature] def getInOutCols: (Array[String], Array[String]) = {
-    require((isSet(inputCol) && isSet(outputCol) && !isSet(inputCols) && !isSet(outputCols)) ||
-      (!isSet(inputCol) && !isSet(outputCol) && isSet(inputCols) && isSet(outputCols)),
-      "QuantileDiscretizer only supports setting either inputCol/outputCol or" +
-        "inputCols/outputCols."
-    )
+  @Since("1.6.0")
+  override def transformSchema(schema: StructType): StructType = {
+    ParamValidators.checkSingleVsMultiColumnParams(this, Seq(outputCol),
```

Comment:
`checkSingleVsMultiColumnParams` can used like `ParamValidators.checkSingleVsMultiColumnParams(this, Seq(outputCol, splits), Seq(outputCols, splitsArray))`.

If we want `numBuckets ` and `numBucketsArray` to be exclusively set, you can use `checkSingleVsMultiColumnParams` like that.

Reply:
Thanks @viirya for your quick reply!
The reason I didn't use 
```
    ParamValidators.checkSingleVsMultiColumnParams(this, Seq(outputCol, numBuckets),
      Seq(outputCols, numBucketsArray))
```
is that we can actually setNumBuckets for multi columns. I looked the previous conversion, we have decided to allow setNumBuckets for multi columns. In the multi columns case
```
If however the numBucketsArray param is unset but the numBuckets param is set, 
the user is saying they want the same numBuckets across all columns, then we can 
use the multi-column version of approxQuantiles in this case.
```

Reply:
ah, I see. thanks!